tests/test_diffraction_objects.py

```diff
@@ -702,3 +702,62 @@ def test_copy_object(do_minimal):
     do_copy = do.copy()
     assert do == do_copy
     assert id(do) != id(do_copy)
+
+
+@pytest.mark.parametrize(
+    "starting_all_arrays, scalar_value, expected_all_arrays",
+    [
+        # Test scalar addition to xarray values (q, tth, d) and expect no change to yarray values
+        (  # C1: Add integer of 5, expect xarray to increase by by 5
+            np.array([[1.0, 0.51763809, 30.0, 12.13818192], [2.0, 1.0, 60.0, 6.28318531]]),
+            5,
+            np.array([[1.0, 5.51763809, 35.0, 17.13818192], [2.0, 6.0, 65.0, 11.28318531]]),
+        ),
+        (  # C2: Add float of 5.1, expect xarray to be added by 5.1
+            np.array([[1.0, 0.51763809, 30.0, 12.13818192], [2.0, 1.0, 60.0, 6.28318531]]),
+            5.1,
+            np.array([[1.0, 5.61763809, 35.1, 17.23818192], [2.0, 6.1, 65.1, 11.38318531]]),
+        ),
+    ],
+)
+def test_addition_operator_by_scalar(starting_all_arrays, scalar_value, expected_all_arrays, do_minimal_tth):
+    do = do_minimal_tth
+    assert np.allclose(do.all_arrays, starting_all_arrays)
+    do_sum = do + scalar_value
+    assert np.allclose(do_sum.all_arrays, expected_all_arrays)
+
```

Comment:
We also need to test the right add, `scalar_value + do`

Reply:
done